backend/internal/kyc_client.go

```diff
@@ -0,0 +1,139 @@
+package internal
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/vedhavyas/go-subkey"
+	"github.com/vedhavyas/go-subkey/sr25519"
+)
+
+// KYCClient holds configuration for tf-kyc-verifier API client
+type KYCClient struct {
+	APIURL          string
+	SponsorAddress  string
+	SponsorKeyPair  subkey.KeyPair
+	ChallengeDomain string
+	HTTPClient      *http.Client
+}
+
+// NewKYCClient creates a new KYCClient instance
+func NewKYCClient(apiURL, sponsorAddress, sponsorPhrase, challengeDomain string) (*KYCClient, error) {
+	var kr subkey.KeyPair
+	var err error
+	if sponsorPhrase != "" {
+		kr, err = sr25519.Scheme{}.FromPhrase(sponsorPhrase, "")
+		if err != nil {
+			return nil, fmt.Errorf("failed to create sponsor keypair: %w", err)
+		}
+	}
+	return &KYCClient{
+		APIURL:          apiURL,
+		SponsorAddress:  sponsorAddress,
+		SponsorKeyPair:  kr,
+		ChallengeDomain: challengeDomain,
+		HTTPClient:      &http.Client{Timeout: 10 * time.Second},
+	}, nil
+}
+
+// createChallengeMessage creates the challenge message string
+func (c *KYCClient) createChallengeMessage() string {
+	timestamp := time.Now().Unix()
+	return fmt.Sprintf("%s:%d", c.ChallengeDomain, timestamp)
+}
+
+// signMessage signs the message with given keypair and returns hex encoded signature
+func signMessage(kr subkey.KeyPair, message string) (string, error) {
+	sig, err := kr.Sign([]byte(message)) // Sign raw bytes, not hash
+	if err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(sig), nil
+}
+
+// CreateSponsorship creates a sponsorship between sponsor and sponsee addresses
+func (c *KYCClient) CreateSponsorship(sponseeAddress string, sponseeKeyPair subkey.KeyPair) error {
+	// Create challenge messages
+	sponsorChallenge := c.createChallengeMessage()
+	sponseeChallenge := c.createChallengeMessage()
+
+	// Sign challenges
+	sponsorSignature, err := signMessage(c.SponsorKeyPair, sponsorChallenge)
+	if err != nil {
+		return fmt.Errorf("failed to sign sponsor challenge: %w", err)
+	}
+	sponseeSignature, err := signMessage(sponseeKeyPair, sponseeChallenge)
+	if err != nil {
+		return fmt.Errorf("failed to sign sponsee challenge: %w", err)
+	}
+
+	// Debug logs for troubleshooting
+	log.Debug().Msgf("KYC Sponsorship Debug: sponsorAddress=%s, sponseeAddress=%s", c.SponsorAddress, sponseeAddress)
+	log.Debug().Msgf("KYC Sponsorship Debug: sponsorChallenge=%s, sponseeChallenge=%s", sponsorChallenge, sponseeChallenge)
+	log.Debug().Msgf("KYC Sponsorship Debug: sponsorSignature=%s, sponseeSignature=%s", sponsorSignature, sponseeSignature)
+
+	// Prepare HTTP request
+	url := fmt.Sprintf("%s/api/v1/sponsorships", c.APIURL)
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer([]byte{}))
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
```

Comment:
Add more context to mkae the error here more useful when spoted


```suggestion
		return fmt.Errorf("creating HTTP request to %s: %w", url, err)
```